lightning/src/ln/peer_handler.rs

```diff
@@ -1209,7 +1216,8 @@ mod tests {
 		assert_eq!(peers[0].peers.lock().unwrap().peers.len(), 1);
 
 		// Since timer_tick_occured() is called again when awaiting_pong is true, all Peers are disconnected
-		peers[0].timer_tick_occured();
-		assert_eq!(peers[0].peers.lock().unwrap().peers.len(), 0);
+		// TODO: simulate handshake completion to fully support ping exchange simulations
```

Comment:
Why not just exchange the handshake? Shouldn't be that hard?

Reply:
I don't think it's super trivial because the test needs to manually call the read events, but I'll give it a try. I originally thought it might be best reserved for a separate diff.

Reply:
This is proving to be mutex mess. I think it will be easier once one of (or preferably both) the modular handshake and the peer handler refactor into individual peer holders commits land. CC: @jkczyz

Reply:
I don't have strong feelings about maintaining this test. It should be a few tests covering the different behaviors. Currently, it tests a subset of the behaviors and so will test a smaller subset when removing this part. It would probably be easier to test after refactoring, as you said.

Reply:
Hmm, should be doable. Can you try building on the top commit on https://github.com/TheBlueMatt/rust-lightning/commits/2020-02-peer_handler-handshake

Reply:
wow, I had no idea you could use the descriptors for that

Reply:
Descriptors, ultimately, are an API for the PeerHandler to send data to a given socket. So that is, in fact, the *only* thing you can use the descriptors for :p

Reply:
Interesting. The word `descriptor` suggested to me that it's just something that describes the socket, aka to be used as a key to find it in the map.